src/api/transform/openai-format.ts

```diff
@@ -1,11 +1,51 @@
 import { Anthropic } from "@anthropic-ai/sdk"
 import OpenAI from "openai"
 
+/**
+ * Normalizes a tool call ID to be compatible with providers that have strict ID requirements.
+ * Some providers (like Mistral) require tool call IDs to be:
+ * - Only alphanumeric characters (a-z, A-Z, 0-9)
+ * - Exactly 9 characters in length
+ *
+ * This function extracts alphanumeric characters from the original ID and
+ * pads/truncates to exactly 9 characters, ensuring deterministic output.
+ *
+ * @param id - The original tool call ID (e.g., "call_5019f900a247472bacde0b82" or "toolu_123")
+ * @returns A normalized 9-character alphanumeric ID
+ */
+export function normalizeToolCallId(id: string): string {
+	// Extract only alphanumeric characters
+	const alphanumeric = id.replace(/[^a-zA-Z0-9]/g, "")
+
+	// Take first 9 characters, or pad with zeros if shorter
+	if (alphanumeric.length >= 9) {
+		return alphanumeric.slice(0, 9)
+	}
+
+	// Pad with zeros to reach 9 characters
+	return alphanumeric.padEnd(9, "0")
+}
+
+/**
+ * Options for converting Anthropic messages to OpenAI format.
+ */
+export interface ConvertToOpenAiMessagesOptions {
+	/**
+	 * The model ID being used. If it contains "mistral", tool call IDs will be
+	 * normalized to be compatible with Mistral's strict ID requirements.
+	 */
+	modelId?: string
```

Comment:
Nitpick: It feels little bit like a "separation of concerns" violation for this module to know that Mistral models require a specific normalization format.

I think a better design would be to pass an optional normalization function in (`undefined` by default), so that the consumer of `convertToOpenAiMessages` can flexibly declare the tool call ID format and presumably the caller will have more information about the provider and model).

Does that make sense?

I know the current implementation might be easier and it gets the job done, but I think this type of edge case handling is already a bit unwieldy throughout our provider code.